src/internal/types.test.ts

```diff
@@ -156,3 +161,144 @@ describe("IfBoundedRecord", () => {
     >().toEqualTypeOf<false>();
   });
 });
+
+describe("EnumerableStringKeyOf", () => {
+  test("string keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<string, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("number keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<number, unknown>>
+    >().toEqualTypeOf<`${number}`>();
+  });
+
+  test("union of records", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<number, unknown> | Record<string, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("union keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<number | string, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("symbol keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<string | symbol, unknown>>
+    >().toEqualTypeOf<string>();
+
+    expectTypeOf<
+      EnumerableStringKeyOf<{ [SymbolFoo]: number; a: unknown }>
+    >().toEqualTypeOf<"a">();
+
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<string | typeof SymbolFoo, unknown>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("optional keys", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<{ a: unknown; b?: unknown }>
+    >().toEqualTypeOf<"a" | "b">();
+  });
+
+  test("branded types", () => {
+    expectTypeOf<
+      EnumerableStringKeyOf<Record<Branded<string, symbol>, unknown>>
+    >().toEqualTypeOf<`${Branded<string, symbol>}`>();
+  });
+});
+
+describe("EnumerableStringKeyedValueOf", () => {
+  test("string values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<number, string>>
+    >().toEqualTypeOf<string>();
+  });
+
+  test("number values", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<Record<string, number>>
+    >().toEqualTypeOf<number>();
+  });
+
+  test("union of records", () => {
+    expectTypeOf<
+      EnumerableStringKeyedValueOf<
+        Record<number, string> | Record<string, string>
```

Comment:
Similar to my comment in one of the specific tests, this doesn't test the union logic very well because both records use the same value type. Instead, we should test two disjoint value types.
```suggestion
        Record<number, "cat"> | Record<string, "dog">
```